Tests/MongoSwiftTests/ConnectionStringTests.swift

```diff
@@ -63,37 +63,31 @@ private struct TestCredential: Decodable {
 // Tests we skip because we don't support the specified behavior.
 let skipUnsupported: [String: [String]] = [
     "valid-auth.json": ["mongodb-cr"], // we don't support MONGODB-CR authentication
+    "compression-options.json": ["multiple compressors are parsed correctly"], // requires Snappy, see SWIFT-894
     "connection-pool-options.json": [
-        // we don't support maxIdleTimeMS.
+        // we don't support maxIdleTimeMS
         "too low maxidletimems causes a warning",
         "non-numeric maxidletimems causes a warning",
         "valid connection pool options are parsed correctly",
-        // We don't support minPoolSize.
+        // we don't support minPoolSize
         "minpoolsize=0 does not error",
-        // We don't allow maxPoolSize=0, see SWIFT-1339.
+        // we don't allow maxPoolSize=0, see SWIFT-1339
         "maxpoolsize=0 does not error"
     ],
-    "connection-options.json": ["*"], // requires maxIdleTimeMS
-    "compression-options.json": ["*"], // requires Snappy, see SWIFT-894
-    "valid-db-with-dotted-name.json": ["*"] // libmongoc doesn't allow db names in dotted form in the URI
+    "connection-options.json": [
+        "valid connection and timeout options are parsed correctly" // we don't support maxIdleTimeMS
```

Comment:
even though we don't support maxIdleTimeMS in the driver, it is a valid connection string option, so do we want to support it in `MongoConnectionString`? Ditto for the other options we don't support.

Reply:
I think it might be confusing/misleading to users if they set an option and receive no error but the option isn't applied anywhere in the driver. Ideally this would be a nice place to log a warning, but absent that ability it could be deceiving to ignore these options silently. This does harm the portability of connection strings, though, which might be more bothersome to users. What do you think @kmahar?

Reply:
@patrickfreed, what are you thinking the use case is for supporting these? portability?

> I think it might be confusing/misleading to users if they set an option and receive no error but the option isn't applied anywhere in the driver

I agree this is definitely undesirable; we need to communicate to the user at some point in some way that we're ignoring these.

to me the options seem to be: 
1) error upon `MongoConnectionString` initialization
2) error upon `MongoClient` initialization `MongoConnectionString` that contains one of these options. this might be technically more correct since they are valid options in general, but is probably equally annoying in practice for users as option 1, and for options we never intend to support (for example single threaded options) it may be confusing to users. 

I'm not sure if this type will have value for users outside of the context of creating a client so I'm not sure if there is a meaningful distinction between where the validation happens, but to avoid confusion I'd probably lean toward option 1

Reply:
It's pretty niche but I guess there could be users who want to write a tool for building or parsing connection strings but not actually connecting to a database. 

More generally speaking, I was wondering if we wanted `MongoConnectionString` to be a general purpose connection string model that can be used for `MongoClient` configuration or a configuration tool for `MongoClient` that can be used to model a connection string (basically 2 vs 1).

Option 2 feels "cleaner" to me in an abstract sense, but practically speaking 1 is probably better. Either way they both seem fine to me, so happy to defer to your all's opinions on this. Just figured it was worth discussing.

Reply:
makes sense! yeah, I think for now I prefer option 1, but we could always move toward the general purpose model in the future in a non-breaking way if that comes up as a desired use case.

Reply:
yeah I'm fine with sticking with the current functionality of erroring upon connection string initialization